### PR TITLE
feat: add arxiv toolkit

### LINKED ISSUE #888: [Feature Request] Load data from Arxiv / google scholar
### Required prerequisites

- [X] I have searched the [Issue Tracker](https://github.com/camel-ai/camel/issues) and [Discussions](https://github.com/camel-ai/camel/discussions) that this hasn't already been reported. (+1 or comment there if it has.)
- [ ] Consider asking first in a [Discussion](https://github.com/camel-ai/camel/discussions/new).

### Motivation

Load academic papers for RAG

### Solution

_No response_

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Wendong-Fan**
## Description

Add arxiv toolkit, including `search_papers` and `download_papers`

## Motivation and Context

 `close #888 ` 

- [x] I have raised an issue to propose this change ([required](https://github.com/camel-ai/camel/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)

## Implemented Tasks

- [ ] Subtask 1
- [ ] Subtask 2
- [ ] Subtask 3

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [ ] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by Wendong-Fan** [comment]
didn't add arxiv to dependency due to

```
Because agentops (0.3.11) depends on requests (2.31.0)
 and agentops (0.3.10) depends on requests (2.31.0), agentops (0.3.10 || 0.3.11) requires requests (2.31.0).
And because agentops (0.3.7) depends on requests (2.31.0), agentops (0.3.7 || 0.3.10 || 0.3.11) requires requests (2.31.0).
And because agentops (0.3.6) depends on requests (2.31.0)
 and no versions of agentops match >0.3.6,<0.3.7 || >0.3.7,<0.3.9 || >0.3.9,<0.3.10 || >0.3.10,<0.3.11 || >0.3.11,<0.3.12 || >0.3.12,<0.4.0, agentops (>=0.3.6,<0.3.9 || >0.3.9,<0.3.12 || >0.3.12,<0.4.0) requires requests (2.31.0).
And because agentops (0.3.12) depends on requests (2.31.0)
 and agentops (0.3.9) depends on requests (2.31.0), agentops (>=0.3.6,<0.4.0) requires requests (2.31.0).
Because no versions of arxiv match >2.1.3,<3.0.0
 and arxiv (2.1.3) depends on requests (>=2.32.0,<2.33.0), arxiv (>=2.1.3,<3.0.0) requires requests (>=2.32.0,<2.33.0).
Thus, arxiv (>=2.1.3,<3.0.0) is incompatible with agentops (>=0.3.6,<0.4.0).
So, because camel-ai depends on both agentops (^0.3.6) and arxiv (^2.1.3), version solving failed.
```

File: camel/toolkits/arxiv_toolkit.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        papers_data = []
+
+        for paper in search_results:
+            paper_info = {
+                "title": paper.title,
+                "published_date": paper.updated.date().isoformat(),
+                "authors": [author.name for author in paper.authors],
+                "entry_id": paper.entry_id,
+                "summary": paper.summary,
+                "paper_text": arxiv_to_text(paper.pdf_url),
```
Comment (lightaime) [inline camel/toolkits/arxiv_toolkit.py:101]: Should we use `unstructured io` instead for better performance?

**Review by lightaime — APPROVED** [review]
Looks good overall. Can we add an example to download a paper like: `attention is all you need`.

File: camel/toolkits/arxiv_toolkit.py (hunk 1 above)
Comment (Wendong-Fan) [reply camel/toolkits/arxiv_toolkit.py:101]: Hey @lightaime , after conducting a comparison test between `unstructured io` and `arxiv_to_text`, I found that `unstructured io` struggles with papers that have content formatted in two columns on a single page, as it ends up scrambling the text. However, `arxiv_to_text` handles this type of formatting well.

Screenshot 2024-10-08 at 14 25 25


Screenshot 2024-10-08 at 14 28 22

**Pull request merged**